server/src/main/java/org/tctalent/server/service/db/aws/S3ResourceHelper.java

```diff
@@ -34,292 +51,285 @@
+    amazonS3.deleteObject(s3Bucket, key);
+  }
+
+  public void copyObject(String sourceKey,
+      String destinationKey) {
+    CopyObjectRequest copyObjectRequest = new CopyObjectRequest(s3Bucket, sourceKey, s3Bucket, destinationKey);
+    transferManager.copy(copyObjectRequest);
+  }
+
+  public void copyObject(String sourceBucket,
+      String sourceKey,
+      String destinationBucket,
+      String destinationKey) {
+
+    sourceBucket = sourceBucket != null ? sourceBucket : s3Bucket;
+    destinationBucket = destinationBucket != null ? destinationBucket : s3Bucket;
+    CopyObjectRequest copyObjectRequest = new CopyObjectRequest(sourceBucket, sourceKey, destinationBucket,
+        destinationKey);
+    transferManager.copy(copyObjectRequest);
+  }
+
+  public void copyBucketContents(String sourceBucket,
+      String sourcePrefix,
+      String destinationBucket,
+      String destinationPrefix)
+      throws AmazonS3Exception {
+    ListObjectsRequest req = new ListObjectsRequest();
+    req.setBucketName(sourceBucket);
+    req.withPrefix(sourcePrefix);
+    List<S3ObjectSummary> contents = amazonS3.listObjects(req).getObjectSummaries();
+    if (contents.isEmpty()) {
+      throw new AmazonS3Exception("No files found under " + sourceBucket + " and prefix " + sourcePrefix);
     }
-
-    public List<S3ObjectSummary> filterMigratedObjects(List<S3ObjectSummary> objectSummaries) {
-        List<S3ObjectSummary> filteredSummaries = objectSummaries.stream().filter(s -> !s.getKey().contains("/migrated/"))
-                .collect(Collectors.toList());
-        return filteredSummaries;
+    for (S3ObjectSummary summary : contents) {
+      String newKey = destinationPrefix + summary.getKey().replace(sourcePrefix, "");
+      CopyObjectRequest copyObjectRequest = new CopyObjectRequest(sourceBucket, summary.getKey(),
+          destinationBucket, newKey);
+      transferManager.copy(copyObjectRequest);
     }
-
-    public void addObjectMetadata(S3ObjectSummary objectSummary) {
-        ObjectMetadata metadata = new ObjectMetadata();
-        Mimetypes mimetypes = Mimetypes.getInstance();
-        metadata.setContentType(mimetypes.getMimetype(objectSummary.getKey()));
-        final CopyObjectRequest request = new CopyObjectRequest(objectSummary.getBucketName(), objectSummary.getKey(), objectSummary.getBucketName(), objectSummary.getKey())
-                .withSourceBucketName(objectSummary.getBucketName())
-                .withSourceKey(objectSummary.getKey())
-                .withNewObjectMetadata(metadata);
-        amazonS3.copyObject(request);
+  }
+
+  public File getTmpFile() throws IOException {
+    File tmpFile = File.createTempFile("lola-", ".tmp");
+    tmpFile.deleteOnExit();
+    return tmpFile;
+  }
+
+  public boolean doesBucketExist(String bucket) {
+    return amazonS3.doesBucketExist(s3Bucket + "bucket");
+  }
+
+  public List<S3ObjectSummary> getObjectSummaries() {
+    ObjectListing objects = amazonS3.listObjects
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).